### PR TITLE
feat(757): Add screwdriver.cd/disk annotation; deprecate beta.screwdriver.cd

### PR DESCRIPTION
**Pull request by Filbird**
## Context
High disk support will soon be available for `k8s-vm`. This is configurable via the `screwdriver.cd/disk` annotation.

Additionally we plan to deprecate `beta.screwdriver.cd` in favor of `screwdriver.cd`

## Objective
Document the `screwdriver.cd/disk` annotation.

Document the deprecation of `beta.screwdriver.cd`.

## References
https://github.com/screwdriver-cd/screwdriver/issues/757
https://github.com/screwdriver-cd/executor-k8s-vm/pull/39
https://github.com/screwdriver-cd/executor-base/pull/48

### CODE REVIEW

File: docs/user-guide/configuration/annotations.md [hunk 1]
```diff
@@ -35,6 +35,7 @@ Some Annotations are used to modify the properties of the build environment. The
 | beta.screwdriver.cd/executor | Ask your cluster admin | This will determine what compute system is used to run the build. For example, set the build to run in a VM, a kubernetes pod, a docker container, or a Jenkins agent. |
 | beta.screwdriver.cd/cpu | `MICRO` / `LOW` / `HIGH` | When using a `k8s` executor, this will allow the user to choose between 0.5 (`MICRO`), 2 (`LOW`) and 6 (`HIGH`) CPU resources. Default is `LOW`.<br>   When using a `k8s-vm` executor, this will allow the user to choose between 1 (`MICRO`), 2 (`LOW`) and 6 (`HIGH`) CPU resources. Default is `LOW`. |
 | beta.screwdriver.cd/ram | `MICRO` / `LOW` / `HIGH` | When using a `k8s` executor, this will allow the user to choose between 1 GB (`MICRO`), 2 GB (`LOW`) and 12 GB (`HIGH`) RAM. Default is `LOW`.<br> When using a `k8s-vm` executor, this will allow the user to choose between 1 GB (`MICRO`), 2 GB (`LOW`) and 12 GB (`HIGH`) RAM. Default is `LOW`. |
+| beta.screwdriver.cd/disk | `LOW` / `HIGH` | When using a `k8s-vm` executor, this will allow the user to choose between 20 GB (`LOW`) and 50 GB (`HIGH`) disk space. Default is `LOW`. |
```
Comment (tkyi) [inline docs/user-guide/configuration/annotations.md]: Not sure if this makes more or less sense, but should this be alphabetically positioned in relation to the other resource annotations? (beta.screwdriver.cd/cpu, beta.screwdriver.cd/disk, beta.screwdriver.cd/ram)

Comment (Filbird) [reply docs/user-guide/configuration/annotations.md]: sounds reasonable!

**Review by tkyi — APPROVED** [review]

**Review by minzcmu — COMMENTED** [review]
Let's change everything to `screwdriver.cd/` and document that `beta.screwdriver.cd` also works but deprecated?

File: docs/user-guide/configuration/annotations.md [hunk 2]
```diff
@@ -23,18 +23,21 @@ jobs:
...
         annotations:
-            beta.screwdriver.cd/cpu: HIGH                      # Use HIGH for CPU
+            screwdriver.cd/cpu: HIGH                      # Use HIGH for CPU
             screwdriver.cd/buildPeriodically: H H(4-7) * * *   # Run the job every day sometime between 4am and 7am UTC.
 ```
 
 ## Supported Annotations
 Some Annotations are used to modify the properties of the build environment. The following annotations are supported by plugins maintained by Screwdriver.cd. Ask your cluster admin which annotations are supported in your Screwdriver cluster.
 
+Please note that `beta.screwdriver.cd` is deprecated in favor of `screwdriver.cd`. However, annotations that are prefixed with `beta.screwdriver.cd` will continue to work.
```
Comment (tkyi) [inline docs/user-guide/configuration/annotations.md]: for now*?

might want to use `>` to highlight it

Example:
>Please note that `beta.screwdriver.cd` is deprecated in favor of `screwdriver.cd`. However, annotations that are prefixed with `beta.screwdriver.cd` will continue to work for now.

**Review by minzcmu — APPROVED** [review]
LGTM but let's merge it after we pull in the parse annotations change internally. So that ppl won't be confused by this when it's not actually available

**Review by tkyi — APPROVED** [review]

**Pull request merged**